server/utils/utils.go

```diff
@@ -0,0 +1,362 @@
+func SplitArgs(s string) ([]string, error) {
+	indexes := regexp.MustCompile("\"").FindAllStringIndex(s, -1)
+	if len(indexes)%2 != 0 {
+		return []string{}, errors.New("quotes not closed")
+	}
+
+	indexes = append([][]int{{0, 0}}, indexes...)
+
+	if indexes[len(indexes)-1][1] < len(s) {
+		indexes = append(indexes, [][]int{{len(s), 0}}...)
+	}
+
+	var args []string
+	for i := 0; i < len(indexes)-1; i++ {
+		start := indexes[i][1]
+		end := Min(len(s), indexes[i+1][0])
+
+		if i%2 == 0 {
+			args = append(args, strings.Split(strings.Trim(s[start:end], " "), " ")...)
+		} else {
+			args = append(args, s[start:end])
+		}
+	}
+
+	cleanedArgs := make([]string, len(args))
+	count := 0
+
+	for _, arg := range args {
+		if arg != "" {
+			cleanedArgs[count] = strings.TrimSpace(arg)
+			count++
+		}
+	}
+
+	return cleanedArgs[0:count], nil
+}
+
+func GetPluginKey() string {
+	var regexNonAlphaNum = regexp.MustCompile("[^a-zA-Z0-9]+")
+	return "mattermost_" + regexNonAlphaNum.ReplaceAllString(GetSiteURL(), "_")
+}
+
+func GetPluginURLPath() string {
+	return "/plugins/" + config.PluginName + "/api/v1"
+}
+
+func GetPluginURL() string {
+	return strings.TrimRight(GetSiteURL(), "/") + GetPluginURLPath()
+}
+
+func GetSiteURL() string {
+	ptr := config.Mattermost.GetConfig().ServiceSettings.SiteURL
+	if ptr == nil {
+		return ""
+	}
+	return *ptr
+}
+
+func GetAtlassianConnectURLPath() string {
+	return "/atlassian-connect.json?secret=" + url.QueryEscape(config.GetConfig().Secret)
+}
+
+func GetConfluenceServerWebhookURLPath() string {
+	return "/server/webhook?secret=" + url.QueryEscape(config.GetConfig().Secret)
+}
+
+func IsSystemAdmin(userID string) bool {
+	user, appErr := config.Mattermost.GetUser(userID)
+	if appErr != nil {
+		return false
+	}
+	return user.IsInRole(model.SystemAdminRoleId)
+}
+
+func Deduplicate(a []string) []string {
+	check := make(map[string]int)
+	result := make([]string, 0)
+	for _, val := range a {
+		check[val] = 1
+	}
+
+	for key := range check {
+		result = append(result, key)
+	}
+
+	return result
+}
+
+type ConfluenceStatus struct {
+	State string `json:"state"`
+}
+
+func NormalizeConfluenceURL(confluenceURL string) (string, error) {
+	u, err := url.Parse(confluenceURL)
+	if err != nil {
+		return "", err
+	}
+	if u.Host == "" {
+		ss := strings.Split(u.Path, "/")
+		if len(ss) > 0 && ss[0] != "" {
+			u.Host = ss[0]
+			u.Path = path.Join(ss[1:]...)
+		}
+		u, err = url.Parse(u.String())
+		if err != nil {
+			return "", err
+		}
+	}
+	if u.Host == "" {
+		return "", errors.Errorf("Invalid URL, no hostname: %q", confluenceURL)
+	}
+	if u.Scheme == "" {
+		u.Scheme = "https"
+	}
+
+	confluenceURL = strings.TrimSuffix(u.String(), "/")
+	return confluenceURL, nil
+}
+
+// CheckConfluenceURL checks if `/status` endpoint of the Confluence URL is accessible
+// and responding with the correct state which is "RUNNING"
+func CheckConfluenceURL(mattermostSiteURL, confluenceURL string, requireHTTPS bool) (_ string, err error) {
```

Comment:
I think these Confluence methods are not "utils" but could be part of a service where we can control more logic and move some of the logic from the commands there to control the logic better and be able to unittest (see my comments in the install command)